internal/addrs/action.go

```diff
@@ -327,3 +329,22 @@ func (a ConfigAction) UniqueKey() UniqueKey {
 type configActionKey string
 
 func (k configActionKey) uniqueKeySigil() {}
+
+// ActionInvocation describes the invocation of an action as part of a plan / apply.
+type ActionInvocation struct {
+	TriggeringResource AbsResourceInstance
+	Action             AbsActionInstance
+	TriggerIndex       int
```

Comment:
Since this is using absolute references, I'd name this something like AbsActionInvocation / ActionInvocationAbs/ whatever. And since you've got the trigger index in here, I'd even consider adding instance: AbsActionInvocationInstance. That name makes it much more obvious to other developers that this is referring to a single specific instance of an {Action & ActionTrigger} 

For more context, I say this because at first I was going to point out that a given action could have multiple triggering resources, which is roughly when I realized that was the point of this struct, and we already have naming conventions that can be repurposed for this context!)

Reply:
Yeah that makes sense, I didn't consider all the other varieties of this to come 👍